### ISSUE TITLE
Provide built-in availability to create a shallow dict

### ISSUE THREAD

**Issue by AlexanderFarkas** [pydantic/pydantic#8787]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

Related [discussion](https://github.com/pydantic/pydantic/discussions/8257)

Since, `_iter` is going to be removed in v3, it's not clear how to create a shallow dict.

Use case:
Unpack pydantic DTO to Domain entity parameters.
```python
def update_with_business_logic(name: Optional[str], description: Optional[str], documents: Optional[DocumentsDTO]):
  ...
  
  
class DocumentsDTO(BaseModel):
  terms_of_service_url: str
  
class PatchDTO(BaseModel):
  name: Optional[str]
  description: Optional[str]
  documents: Optional[DocumentsDTO]
  
def fast_api_route(dto: PatchDTO):
  update_with_business_logic(**dto.shallow_model_dump(exclude_unset=True))
```

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/concepts/models/#basic-model-usage)
- [X] [Data serialization](https://docs.pydantic.dev/concepts/serialization/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/concepts/json_schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/concepts/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/concepts/config/)
- [ ] [Field Types](https://docs.pydantic.dev/api/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/concepts/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/concepts/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/concepts/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by sydney-runkle** [comment]
@AlexanderFarkas,

Thanks for the feature request. The current logic used in `__iter__` is the following:

https://github.com/pydantic/pydantic/blob/5d09b4754d4abcdfefaae554de4dae36124a683c/pydantic/main.py#L963-L968

Even if we do deprecate `__iter__` in V3, I think you could still use `some_model.__dict__` and `some_model.__pydantic_extra__` to get what you're looking for. Let me know if you think otherwise / if I'm missing something regarding what you're asking for!

**Comment by AlexanderFarkas** [comment]
Hey @sydney-runkle,

I haven't tested, but I believe your solution will yield optional keys as None. 
It's not possible to apply `exclude_unset=True` to usual __iter__

**Comment by sydney-runkle** [comment]
@AlexanderFarkas,

Right, but you could write custom logic to filter out `None` values. Closing for now, as I think this is logic that should reside in user-code based on the specific use case.
